R/build_copy_queries.R

```diff
@@ -0,0 +1,131 @@
+
+build_copy_queries <- function(con, dm, set_key_constraints = TRUE, temporary = TRUE, table_names) {
+  ## apply filters
+  dm <- dm_apply_filters(dm)
+
+  ## helper to quote all elements of a column and enumerate (concat) element wise
+  quote_enum_col <- function(x) {
+    map_chr(x, ~toString(map_chr(.x, DBI::dbQuoteIdentifier, con = con)))
```

Comment:
At least one of the `map_chr()` calls might be unnecessary here?

Reply:
Not in general, the outer `map_chr()`  to iterates through rows, the inner `map_chr()` maps through elements contained in the cell which might be a list (not always) and quotes them.